### PR TITLE
[Translator] Add comments when dumping po files

### PR DESCRIPTION
**Pull request by deguif**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | ?    
| Fixed tickets | #29962
| License       | MIT
| Doc PR        | 

This code
```php
$catalogue = new MessageCatalogue('fr');
$dumper = new PoFileDumper();

$catalogue->set('key.one', 'First key', 'custom');
$catalogue->setMetadata('key.one', ['sources' => 'src/file_1', 'comments' => 'Comment', 'flags' => 'fuzzy'], 'custom');

$catalogue->set('key.second', 'Second key', 'custom');
$catalogue->setMetadata('key.second', ['sources' => ['src/file_1', 'src/file_2'], 'comments' => ['Comment 1', 'Comment 2'], 'flags' => ['fuzzy', 'another']], 'custom');

$dumper->dump($catalogue, [
    'path' => 'xxx',
]);
```

Will produces this output:
```
msgid ""
msgstr ""
"Content-Type: text/plain; charset=UTF-8\n"
"Content-Transfer-Encoding: 8bit\n"
"Language: fr\n"

# Comment
#, fuzzy
#: src/file_1
msgid "key.one"
msgstr "First key"

# Comment 1
# Comment 2
#, fuzzy,another
#: src/file_1 src/file_2
msgid "key.second"
msgstr "Second key"
```

### CODE REVIEW

**Comment by Nyholm** [comment]
I like this PR. Thank you 
👍 

--------------

Just and FYI: 

In xliff we have `notes` and `target-attributes`. Here we introduce `sources`, `comments` and `flags`. 

We maybe should have 4 standard thing keys that are shared over all dumpers/loaders.

**Comment by fabpot** [comment]
Looks like the tests are broken by the features introduced in this PR. @deguif Can you have a look?

**Comment by deguif** [comment]
Please don't merge, I will rebase it once #30919 will get merged.

**Comment by fabpot** [comment]
@deguif You can rebase this one now.

**Comment by deguif** [comment]
@fabpot this should be mergeable, test failures are not related.

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @deguif.

**Pull request merged**
